android/modules/geolocation/src/java/ti/modules/titanium/geolocation/GeolocationModule.java

```diff
@@ -762,7 +767,36 @@ public boolean getLocationServicesEnabled()
 												   @Kroll.argument(optional = true) final KrollFunction callback)
 	{
 		return KrollPromise.create((promise) -> {
-			tiLocation.forwardGeocode(address, createGeocodeResponseHandler(callback, promise));
+			final KrollDict response = new KrollDict();
+
+			response.put(TiC.EVENT_PROPERTY_SOURCE, this);
+
+			try {
+				final List<Address> addresses = geocoder.getFromLocationName(address, 1);
```

Comment:
The `Geocoder.getFromLocationName()` blocks the current thread. Supposedly if you have network access but no Internet access, then this will block until it times-out. So, we'll need to do this on another thread.